File: Leetcode/python/sort-jumbled-numbers.py
from typing import List

class Solution:
    def sortJumbled(self, mapping: List[int], nums: List[int]) -> List[int]:
        pairs = []

        for i, n in enumerate(nums):
            n = str(n)
            mappedN = 0

            for c in n:
                mappedN *= 10
                mappedN += mapping[int(c)]
            pairs.append((mappedN, i))

        pairs.sort()

        return [nums[p[1]] for p in pairs]

File: Leetcode/typescript/sort-jumbled-numbers.ts
function sortJumbled(mapping: number[], nums: number[]): number[] {
  let pairs: [number, number][] = [];

  for (let i = 0; i < nums.length; i++) {
    const numStr = `${nums[i]}`;
    let mappedNum = 0;

    for (const c of numStr) {
      mappedNum = mappedNum * 10 + mapping[parseInt(c)];
    }
    pairs.push([mappedNum, i]);
  }
  pairs.sort((a, b) => a[0] - b[0] || a[1] - b[1]);
  return pairs.map((pair) => nums[pair[1]]);
}
